### PR TITLE
New module statvfs: retrieve various information on a FS

### PR DESCRIPTION
**Pull request by Yannig**
This module retrieve information about a FS to check if there is enough room for installation, FS size.

Here's a simple playbook using this new module:

``` yaml

---
- name: "Lookup test"
  hosts: "localhost"
  tasks:
    - name: "check / FS"
      statvfs: path=/ unit=G
      register: vfs
      failed_when: vfs.statvfs.used_space > 100
    - debug: var=vfs
    - assert:
        that: 'vfs.statvfs.free_space_pc < 100'
    - name: Check /tmp for available space
      statvfs: path=/tmp unit=M
      register: tmp_info
      failed_when: 'tmp_info.statvfs.free_space < 100'
```

Result:

``` yaml
ansible-playbook test/integration/test_statvfs.yml 

PLAY [Lookup test] ************************************************************ 

GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [check / FS] ************************************************************ 
ok: [localhost]

TASK: [debug var=vfs] ********************************************************* 
ok: [localhost] => {
    "var": {
        "vfs": {
            "changed": false, 
            "failed": false, 
            "failed_when_result": false, 
            "invocation": {
                "module_args": "path=/ unit=G", 
                "module_name": "statvfs"
            }, 
            "statvfs": {
                "free_space": 95.27, 
                "free_space_pc": 86.68, 
                "fs_block_avail": 24975224, 
                "fs_block_count": 28812032, 
                "fs_block_free": 26419483, 
                "fs_block_reserved": 1444259, 
                "fs_block_size": 4096, 
                "fs_block_used": 2392549, 
                "fs_free_inode_pc": 96.47, 
                "fs_inode_avail": 7072771, 
                "fs_inode_count": 7331840, 
                "fs_inode_free": 7072771, 
                "fs_inode_used": 259069, 
                "fs_reserved_inode": 0, 
                "fs_used_inode_pc": 3.53, 
                "reserved_space": 5.51, 
                "total_space": 109.91, 
                "unit": "G", 
                "unit_size": 1073741824, 
                "used_space": 9.13, 
                "used_space_pc": 8.3
            }
        }
    }
}

TASK: [assert ] *************************************************************** 
ok: [localhost]

TASK: [Check /tmp for available space] **************************************** 
ok: [localhost]

PLAY RECAP ******************************************************************** 
localhost                  : ok=5    changed=0    unreachable=0    failed=0
```

### CODE REVIEW

**Comment by bcoca** [comment]
would this make sense more as a facts module?

just rename to stafvfs_facts and return the data into ansible_facts

**Comment by gregdek** [comment]
@Yannig can you make the recommended change and request re-review?

**Comment by Yannig** [comment]
@gregdek  : Yes I can do it. I was just a little bit dutty last time.

**Comment by Yannig** [comment]
Here is a playbook that can be used to check the module:

``` yaml
---
- name: "Lookup test"
  hosts: "localhost"
  tasks:
    - name: "check / FS"
      statvfs_facts: path=/ unit=G
      failed_when: statvfs.used_space < 0
    - assert:
        that: 'statvfs.free_space_pc < 100'
    - name: "Check /tmp for available space"
      statvfs_facts: path=/tmp unit=M
    - assert:
        that: 'statvfs.free_space < 1000'
```

**Comment by gregdek** [comment]
@bcoca can you re-review?

File: files/statvfs_facts.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+'''
+
+import os
+import sys
+import pwd
+import grp
+
+def main():
+    # Size convertion
+    size_convertion = {
```
Comment (bcoca) [inline files/statvfs_facts.py:62]: module.pretty_bytes function might work here, no need to specify size, the most appropriate will be used (sorry I missed this the first time). Though we might want to return the original and human readable side by side.

**Comment by bcoca** [comment]
actually, we are already doing statvfs in the general fact gathering, calling it per mount point found, I see this has a bit more data, should we just add it there?

**Comment by Yannig** [comment]
@bcoca: I have included fact gathering in a new pull request (https://github.com/ansible/ansible/pull/11896)

**Pull request closed without merging**